ctapipe/image/time_next_neighbor_cleaning.py

```diff
@@ -0,0 +1,742 @@
+        """
+        Get or set the current _neighbor_group. Setting the _neighbor_group to
+        a new value also changes `_multiplicity' to the corresponding value.
+        """
+        return self.__neighbor_group
+
+    @_neighbor_group.setter
+    def _neighbor_group(self, nfold):
+        self.__neighbor_group = nfold
+        self._multiplicity = self._multiplicity_dict[nfold]
+
+    def load_individual_pixel_rates(self, file):
+        """
+        Load the IPR graphs from a json file. The dictionaries should contain
+        one dictionary for each camera_id with the pixel rates and charges.
+
+        Parameters
+        ----------
+        file: string
+            json file in which the IPR graphs is stored
+        """
+        with open(file, "r") as f:
+            self.IPR_dict = json.load(f)
+
+    def dump_individual_pixel_rates(self, file):
+        """
+        Dump the IPR graph to file using json.
+
+        Parameters
+        ----------
+        file: string
+
+        """
+        with open(file, "w") as f:
+            json.dump(self.IPR_dict, f, sort_keys=True, indent=4)
+
+    def fill_individual_pixel_rate(self, geo, image, trace_length, charge_steps):
+        """
+        Count the number of pixels in images about a given charge threshold
+        and fill dictionary with individual pixel noise rate (IPR).
+
+        Parameters
+        ----------
+        geo: ctapipe.instrument.CameraGeometry
+        image: numpy.ndarray
+        trace_length: astropy.units.Quantity
+            total length of the trace, e.g. number of samples times sample_time
+        charge_steps: numpy.ndarray
+            array with the charge steps used to calculate the IPR graphs.
+        """
+
+        self.IPR_dict = self.IPR_dict if self.IPR_dict else {}
+
+        if geo.camera_name not in self.IPR_dict:
+            self.IPR_dict[geo.camera_name] = {
+                "counter": 0,
+                "charge": charge_steps,
+                "npix": np.zeros(charge_steps.shape[0]),
+                "rate": np.zeros(charge_steps.shape[0]),
+                "rate_err": np.zeros(charge_steps.shape[0]),
+            }
+
+        # count number of pixels
+        self.IPR_dict[geo.camera_name]["counter"] += np.sum(image >= 0)
+        for thbin, charge in enumerate(self.IPR_dict[geo.camera_name]["charge"]):
+            self.IPR_dict[geo.camera_name]["npix"][thbin] += np.sum(image > charge)
+
+        # convert count of number of pixels above threshold into rate
+        sample_to_hz = (
+            1 / (trace_length * self.IPR_dict[geo.camera_name]["counter"])
+        ).to("Hz")
+        self.IPR_dict[geo.camera_name]["rate"] = (
+            self.IPR_dict[geo.camera_name]["npix"] * sample_to_hz
+        )
+        self.IPR_dict[geo.camera_name]["rate_err"] = (
+            np.sqrt(self.IPR_dict[geo.camera_name]["npix"]) * sample_to_hz
+        )
+
+    def calculate_ipr_from_source(self, source, calibrator, charge_steps):
```

Comment:
This method is I think not really helpful to have here. We should probably just do a `from_images` that can than work with output from the process tool